server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2635,6 +2637,9 @@ private void innerOpenEngineAndTranslog(LongSupplier globalCheckpointSupplier, b
         // time elapses after the engine is created above (pulling the config settings) until we set the engine reference, during
         // which settings changes could possibly have happened, so here we forcefully push any config changes to the new engine.
         onSettingsChanged();
+        if (indexSettings.getIndexMetadata().useIngestionSource()) {
```

Comment:
a nitpick, lets add this condition to the assert instead of returning early.

Reply:
i think this flow is triggered for ingestion source? but we dont want to run `assertSequenceNumbersInCommit` in the next line, because it fails.